plugins/module_utils/consul.py

```diff
@@ -233,24 +263,33 @@ def create_object(self, obj):
         if self._module.check_mode:
             return obj
         else:
-            return self.put(self.api_endpoint, data=self.prepare_object({}, obj))
+            url = self.endpoint_url(OPERATION_CREATE)
+            created_obj = self.put(url, data=self.prepare_object({}, obj))
+            if created_obj is None:
+                return self.read_object()
```

Comment:
```suggestion
                created_obj = self.read_object()
```
While slightly less efficient, this would be more "stabler" with regards to future changes where you might want to modify the object before returning or so.